govcd/vm_test.go

```diff
@@ -551,6 +551,56 @@ func (vcd *TestVCD) Test_InsertOrEjectMedia(check *C) {
 	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
 }
 
+func (vcd *TestVCD) Test_AddMetadataOnVm(check *C) {
```

Comment:
I recommend not only test that Task completed successful but query vapp metadata and verify is there or not.

Reply:
That's a bit trickier. I'm working on it though. I have to implement a new type `type metadata` which can store vapp and vm metadata since metadata is not part of vapp/vm representation which is sad but that's how it is.

Reply:
Fixed.